src/classes/SubmissionSwagger.ts

```diff
@@ -128,7 +133,35 @@ export class SubmissionSwagger extends ResourceSwagger {
         };
         break;
       case 'datagrid':
-        // TODO: finish datagrid swagger def.
+        utils.eachComponent(component.components, (component: any) => {
```

Comment:
Or just modify `getSubmissionDataSchema` to accept `component`/`this.form` as an input and use that.

Reply:
i.e. like this:
```typescript
const datagrid = this.getSubmissionDataSchema(component)
datagrid['type'] = 'object'
property = {
    type: 'array',
    items: datagrid
};
// Prevent eachComponent from going through these components again
component.components = [];
```